### PR TITLE
fix: Vendor can signup even though store URL not available warning exists

### PR DESCRIPTION
**Pull request by devAsadNur**
### All Submissions:

* [x] My code follows the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [ ] I've included developer documentation (optional)
* [x] I've added proper labels to this pull request

### Changes proposed in this Pull Request:
Now, new vendor won't be able to signup if the store URL is not available.

* Closes: https://github.com/weDevsOfficial/dokan-pro/issues/2212.

### How to test the changes in this Pull Request:

1. Go to `My Account` page.
2. Try to signup as a vendor with an existing store URL.
3. It will be prevented by throwing an alert.

### Changelog entry

> fix: Vendor can signup even though store URL not available warning exists

### PR Self Review Checklist:

* Code is not following code style guidelines
* Bad naming: make sure you would understand your code if you read it a few months from now.
* KISS: Keep it simple, Sweetie (not stupid!).
* DRY: Don't Repeat Yourself.
* Code that is not readable: too many nested 'if's are a bad sign.
* Performance issues
* Complicated constructions that need refactoring or comments: code should almost always be self-explanatory.
* Grammar errors.

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.

> As a contributor, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.

* [ ] Correct — Does the change do what it’s supposed to? ie: code 100% fulfilling the requirements?
* [ ] Secure — Would a nefarious party find some way to exploit this change? ie: everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities?
* [ ] Readable — Will your future self be able to understand this change months down the road?
* [ ] Elegant — Does the change fit aesthetically within the overall style and architecture?

### CODE REVIEW

**Comment by nurul-umbhiya** [comment]
Can we also add js validation for this? Previously validation was working from the js end. Maybe some pr changed that behavior.

**Comment by devAsadNur** [comment]
> Can we also add js validation for this? Previously validation was working from the js end. Maybe some pr changed that behavior.

Bhai, JS validation has been added along with the backend validation.

File: includes/Registration.php [hunk 1]
```diff
@@ -55,14 +55,15 @@ public function validate_registration( $error ) {
...
-
+        $role           = sanitize_text_field( wp_unslash( $_POST['role'] ) );
+        $shop_url       = isset( $_POST['shopurl'] ) ? sanitize_text_field( wp_unslash( $_POST['shopurl'] ) ) : '';
         $required_fields = apply_filters(
             'dokan_seller_registration_required_fields', [
                 'fname'    => __( 'Please enter your first name.', 'dokan-lite' ),
                 'lname'    => __( 'Please enter your last name.', 'dokan-lite' ),
                 'phone'    => __( 'Please enter your phone number.', 'dokan-lite' ),
                 'shopname' => __( 'Please provide a shop name.', 'dokan-lite' ),
+                'shopurl'  => __( 'Please provide an unique shop URL.', 'dokan-lite' ),
```
Comment (nurul-umbhiya) [inline includes/Registration.php:66]: Please provide a unique shop URL.

File: includes/Registration.php [hunk 2]
```diff
@@ -73,6 +74,11 @@ public function validate_registration( $error ) {
                     return new WP_Error( "$field-error", $msg );
                 }
             }
+
+            // Check if the shop URL already not in use.
+            if ( ! empty( get_user_by( 'slug', $shop_url ) ) ) {
+                return new WP_Error( 'shop-url-error', __( 'Shop URL not available', 'dokan-lite' ) );
```
Comment (nurul-umbhiya) [inline includes/Registration.php:80]: The shop URL is not available.

**Review by nurul-umbhiya — APPROVED** [review]

**Pull request merged**